pipenv/cli.py

```diff
@@ -137,10 +138,13 @@ def add_to_path(p):
 @background.task
 def check_for_updates():
     """Background thread -- beautiful, isn't it?"""
+    if not need_update_check():
+        return
     try:
         r = requests.get('https://pypi.python.org/pypi/pipenv/json', timeout=0.5)
         latest = max(map(semver.parse_version_info, r.json()['releases'].keys()))
         current = semver.parse_version_info(__version__)
+        checked_for_updates()
```

Comment:
Can I suggest putting this before the call to `requests.get()`? That way, if the user gets bored of waiting for the update check to time out and hits ctrl-C, the update check won't run for the next 24 hours (by which time hopefully the network will be working again) rather than the update check running (and blocking) every time Pipenv runs.

Reply:
I would agree that this should happen before requests.get()